### ISSUE TITLE
Update major dependencies

### ISSUE THREAD

**Issue by Endeavor275** [TryGhost/Ghost#9389]
### Issue Summary

A few of the dependencies are deprecated and I think they should be updated
```npm WARN deprecated nodemailer@0.7.1: All versions below 4.0.1 of Nodemailer are deprecated. See https://nodemailer.com/status/
npm WARN deprecated coffee-script@1.10.0: CoffeeScript on NPM has moved to "coffeescript" (no hyphen)
npm WARN deprecated coffee-script@1.3.3: CoffeeScript on NPM has moved to "coffeescript" (no hyphen)
npm WARN deprecated minimatch@0.2.14: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated mimelib@0.2.19: This module is deprecated
npm WARN deprecated graceful-fs@1.2.3: please upgrade to graceful-fs 4 for compatibility with current and future versions of Node.js
npm WARN deprecated minimatch@0.3.0: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated node-uuid@1.4.8: Use uuid module instead
```

### Steps to Reproduce

1. Get a new install of Ghost's dev edition
2. Run `npm install`
3. Watch the deprecated messages roll in

### Technical details:

* Ghost Version: 1.20.0
* Node Version: 8.9.4
* Browser/OS: Safari on macOS High Sierra
* Database: MySQL

**Comment by kirrg001** [comment]
> npm WARN deprecated coffee-script@1.10.0: CoffeeScript on NPM has moved to "coffeescript" (no hyphen)
npm WARN deprecated coffee-script@1.3.3: CoffeeScript on NPM has moved to "coffeescript" (no hyphen)

https://github.com/gruntjs/grunt/issues/1611

> npm WARN deprecated mimelib@0.2.19: This module is deprecated

Can't be resolved, because we are stuck with nodemailer 0.7, [see](https://github.com/TryGhost/Ghost/issues/8933).

> npm WARN deprecated minimatch@0.2.14: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
> npm WARN deprecated minimatch@0.3.0: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
> npm WARN deprecated graceful-fs@1.2.3: please upgrade to graceful-fs 4 for compatibility with current and future versions of Node.js

Therefor we have to bump major versions of some dependencies e.g. archiever, glob. Will take a look. Although `nsp check` doesn't show `RegExp DoS`.

> npm WARN deprecated node-uuid@1.4.8: Use uuid module instead

Can be resolved by updating knex + bookshelf.

**Comment by kirrg001** [comment]
We have a couple of major updates outstanding. I will try to bump some next Monday.
